src/components/05_pages/Permissions/Permissions.js

```diff
@@ -1,17 +1,112 @@
-import React from 'react';
-import { css } from 'emotion';
+import makeCancelable from 'makecancelable';
 
-const styles = {
-  title: css`
-    text-decoration: underline;
-  `,
-};
+import React, { Fragment, Component } from 'react';
+import Loading from '../../Helpers/Loading';
 
-const Permissions = () => (
-  <div>
-    <h1 className={styles.title}>Permissions</h1>
-    <p>This will be the permissions page.</p>
-  </div>
-);
+const Permissions = class Permissions extends Component {
+  state = {
+    loaded: false,
+  };
+  componentDidMount() {
+    this.cancelFetch = makeCancelable(
+      Promise.all([
+        fetch(
+          `${process.env.REACT_APP_DRUPAL_BASE_URL}/permissions?_format=json`,
+        ).then(res => res.json()),
+        fetch(
+          `${
+            process.env.REACT_APP_DRUPAL_BASE_URL
+          }/jsonapi/user_role/user_role`,
+        ).then(res => res.json()),
+      ])
+        .then(([permissions, { data: roles }]) =>
+          this.setState({ permissions, roles, loaded: true }),
+        )
+        .catch(err => this.setState({ loaded: false, err })),
+    );
+  }
+  componentWillUnmount() {
+    this.cancelFetch();
+  }
+  fetchData = () =>
+    Promise.all([
+      fetch(
+        `${process.env.REACT_APP_DRUPAL_BASE_URL}/permissions?_format=json`,
+      ).then(res => res.json()),
+      fetch(
+        `${process.env.REACT_APP_DRUPAL_BASE_URL}/jsonapi/user_role/user_role`,
+      ).then(res => res.json()),
+    ]);
+  groupPermissions = permissions =>
+    Object.entries(
+      Object.keys(permissions)
+        .map(key => permissions[key])
+        .reduce((acc, cur) => {
+          acc[cur.provider] = acc[cur.provider] || [];
+          acc[cur.provider].push(cur);
+          return acc;
+        }, {}),
+    );
+  render() {
+    return !this.state.loaded ? (
+      <Loading />
+    ) : (
+      <table>
+        <thead>
+          <tr>
+            {[
+              'PERMISSION',
+              ...this.state.roles.map(({ attributes: { label } }) =>
+                label.toUpperCase(),
+              ),
+            ].map(label => <td key={`column-${label}`}>{label}</td>)}
+          </tr>
+        </thead>
+        <tbody>
+          {this.groupPermissions(this.state.permissions).map(
+            ([permissionGroupName, permissions]) =>
+              permissions.length && (
+                <Fragment key={`fragment-${permissionGroupName}`}>
+                  <tr key={`permissionGroup-${permissionGroupName}`}>
```

Comment:
Do we need key on both levels?

Reply:
Yep. Which is strange, but each element within the loop needs a key.